ci-scripts/linux/local-build/linux_depends.sh

```diff
@@ -0,0 +1,268 @@
+	# Install dependencies for Arch Linux-based systems
+	echo -e "\n\n"
+    for paquetes in dialog base-devel git cmake boost boost-libs qt5-base qt5-svg qt5-script \
+        qt5-tools qt5-multimedia lz4 lzo libjpeg-turbo glew freeglut freetype2 blas cblas \
+        superlu libmypaint; do pacman -S --needed
+    done
+    touch /tmp/ok-packages
+	echo -e "\n\n$(_msg OK_DEPENDS)\n\n"
+}
+
+function _openSuseDepends() {
+	# Install dependencies for openSUSE systems
+    echo -e "\n\n"
+    for paquetes in dialog boost-devel cmake freeglut-devel freetype2-devel gcc-c++ glew-devel \
+        libQt5OpenGL-devel libjpeg-devel liblz4-devel libpng16-compat-devel libqt5-linguist-devel \
+        libqt5-qtbase-devel libqt5-qtmultimedia-devel libqt5-qtscript-devel libqt5-qtsvg-devel \
+        libtiff-devel lzo-devel openblas-devel pkgconfig sed superlu-devel zlib-devel json-c-devel \
+        libqt5-qtmultimedia libmypaint-devel libmypaint libjpeg62-turbo-devel; do zypper install -y
+    done
+    touch /tmp/ok-packages
+    echo -e "\n\n$(_msg OK_DEPENDS})\n\n"
+}
+
+function _verifpackages() {
+	if [ -e "/tmp/ok-packages" ]; then
+	echo -e "\n\n$(_msg OK_DEPENDS)\n\n"
+	exit 1
+fi 
+}
+
+function _warning() {
+    while true; do
+		clear
+        # Show the message and prompt the user for input
+        echo "$(_msg WARNING)"
+        echo "1) $(_msg ACEPT)"
+        echo "2) $(_msg CANCEL)"
+        read -p "$(_msg OPTION):" option
+        
+        # Check the selected option
+        case $option in
+            1)
+                return 0  # Exit the function with success
+                ;;
+            2)
+                echo -e "\n\n$(_msg EXIT_INSTALL)\n\n"
+                exit 1  # Exit the script
+                ;;
+        esac
+    done
+}
+
+
+function _update() {
+
+    # Detect the package manager and update the system
+		if command -v apt-get &>/dev/null; then
+			apt-get update && apt-get upgrade -y
+		elif command -v dnf &>/dev/null; then
+			dnf update -y
+		elif command -v yum &>/dev/null; then
+			yum update -y
+		else
+			echo -e "\n\n$(_msg PACKAGE_GESTOR_NOT_FOUND)\n\n"
+			exit 1
+		fi   
+}
+
+function _depends() {
+	# Detect and install dependencies based on the system's package manager
+	clear	
+	
+		if command -v apt-get >/dev/null 2>&1; then
+			_debianDepends
+
+		elif command -v dnf >/dev/null 2>&1; then
+			_fedoraDepends
+
+		elif command -v pacman >/dev/null 2>&1; then
+			_archDepends
+
+		elif command -v zypper >/dev/null 2>&1; then
+			_openSuseDepends
+			
+		else
+			echo -e "\n\n$(_msg PACKAGE_GESTOR_NOT_FOUND)\n\n"
+			exit 1
+		fi
+}
+
+# ======================================================================
+# Functions execution
+# ======================================================================
+
+_rootcheck      # Check for root privileges
+_verifpackages  # Verify the existence of the test file
+_warning		# Show a warning about process
+_update         # Update the system
```

Comment:
Why would you update the user's system? This seems well out of scope for an installation script.